pandas/tests/io/parser/header.py

```diff
@@ -286,3 +286,10 @@ def test_non_int_header(self):
             self.read_csv(StringIO(data), sep=',', header=['a', 'b'])
         with tm.assert_raises_regex(ValueError, msg):
             self.read_csv(StringIO(data), sep=',', header='string_header')
+
+    def test_singleton_header(self):
+        # See GH #7757
+        data = """a,b,c\n0,1,2\n1,2,3"""
+        df = self.read_csv(StringIO(data), header=[0])
+        expected = self.read_csv(StringIO(data), header=0)
```

Comment:
I would explicitly construct it using the `DataFrame` constructor.  This makes the test more reliable.